### PR TITLE
Do not do operations with pointer to nullptr in image.cpp

### LINKED ISSUE #8285: Undefined behavior in fheroes2::Flip()
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

macOS

### Describe the bug

If output image is single-layer, `transformOutY` becomes `nullptr`:

https://github.com/ihhub/fheroes2/blob/618cfd9c3aa9c8a4748d0f625e4d274a5cd00f42/src/engine/image.cpp#L2263

But later `transformOutY` is subject to arithmetic operations, e.g. here:

https://github.com/ihhub/fheroes2/blob/618cfd9c3aa9c8a4748d0f625e4d274a5cd00f42/src/engine/image.cpp#L2295

which is [undefined behavior](https://en.cppreference.com/w/cpp/language/operator_arithmetic#Pointer_arithmetic) - you only can add or subtract zero to/from `nullptr`, and that's all.

The same problem is probably present in other places where single-layer images are processed.

### Save file

None needed.

### Additional info

Credits: Clang UB Sanitizer.

### PR DESCRIPTION
**Pull request by Districh-ru**
close #8285

### CODE REVIEW

File: src/engine/image.cpp [hunk 1]
```diff
@@ -1412,21 +1423,21 @@ namespace fheroes2
                     --imageOutRightPoint;
                     --imageInRightPoint;
 
-                    if ( transformOutRightPoint ) {
+                    if ( transformOutRightPoint != nullptr ) {
```
Comment (oleg-derevenetz) [inline src/engine/image.cpp:1426]: `if ( transformOutRightPoint ) {` is also OK, [non-null pointers are truthy and null pointers are falsy](https://en.cppreference.com/w/cpp/language/implicit_conversion#Boolean_conversions), so previous syntax is also OK here and in other places.

**Review by oleg-derevenetz — APPROVED** [review]
Hi @Districh-ru I just did a test run of this PR's branch using the same sanitizer, no issues were found. There are some unnecessary (from my point of view) changes, but it's a matter of taste.

File: src/engine/image.cpp (hunk 1 above)
Comment (Districh-ru) [reply src/engine/image.cpp:1426]: Hi @oleg-derevenetz, I made these changes to more clearly show that we are checking for `nullptr`. The compiled code will be the same.

Hi @ihhub, what syntax is better to leave in our code?

Comment (ihhub) [reply src/engine/image.cpp:1426]: We can the way you implemented. It's a matter of taste but your way is not explicit.

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@Districh-ru , thank you very much for these changes!

**Comment by oleg-derevenetz** [comment]
Hi @Districh-ru probably SonarQube analyzer needs some checks or assertions to ensure that there will be no dereferences of null pointers:

https://sonarcloud.io/project/issues?resolved=false&types=BUG&id=ihhub_fheroes2

You can create PR branch directly on this repo to run the SonarQube checks for that PR.

**Comment by Districh-ru** [comment]
> Hi @Districh-ru probably SonarQube analyzer needs some checks or assertions to ensure that there will be no dereferences of null pointers:
> 
> https://sonarcloud.io/project/issues?resolved=false&types=BUG&id=ihhub_fheroes2
> 
> You can create PR branch directly on this repo to run the SonarQube checks for that PR.

Hi @oleg-derevenetz , thanks for noticing this. I'll use SonarLint Visual Studio extension to check these bugs.

**Pull request merged**